app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue

```diff
@@ -72,28 +76,79 @@ export default {
       const arrayColumnsQueryString = this.arrayColumnsQuery.toString()
 
       if (arrayColumnsQueryString !== data) {
-        this.initPerspective(this.items)
+        this.checkPerspectiveTypes()
       } else {
         this.viewer.clear()
         // Well, it's a bit tricky, but reset the table with .clear() only responds when trigger an event, if not trigger an event .clear() isn't fired
         window.dispatchEvent(new Event('resize'))
       }
     },
+    checkPerspectiveTypes() {
+      //Get columns generates from query
+      const arrayColumnsFromQuery = this.arrayColumnsQuery
+      //Get columns from API
+      const arrayColumnsFromAPI = Object.keys(this.objectColumns)
+
+      /* We compare the columns, if it returns true we pass the schema to Perspective, if false Perspective will convert the columns*/
+      const sameColumns = arrayColumnsFromAPI.length == arrayColumnsFromQuery.length && arrayColumnsFromAPI.every(column => arrayColumnsFromQuery.includes(column))
```

Comment:
Use three equals in comparisons

```suggestion
      const sameColumns = arrayColumnsFromAPI.length === arrayColumnsFromQuery.length && arrayColumnsFromAPI.every(column => arrayColumnsFromQuery.includes(column))
```